packages/react-aria-components/src/Table.tsx

```diff
@@ -862,7 +862,9 @@ function TableColumnHeader<T>({column}: {column: GridNode<T>}) {
       ref={ref}
       data-focused={isFocused || undefined}
       data-focus-visible={isFocusVisible || undefined}
-      data-resizing={isResizing || undefined}>
+      data-resizing={isResizing || undefined}
+      data-sorting={column.props.allowsSorting || undefined}
```

Comment:
This one doesn't match the docs above. Maybe `data-allows-sorting` would be a better name?